cpp/ql/src/semmle/code/cpp/models/implementations/Strcat.qll

```diff
@@ -24,6 +24,21 @@ class StrcatFunction extends TaintFunction, DataFlowFunction, ArrayFunction, Sid
     )
   }
 
+  /**
+   * Gets the index of the parameter that is the size of the copy (in characters).
+   */
+  int getParamSize() { if exists(getParameter(2)) then result = 2 else none() }
```

Comment:
Writing this as `exists(getParameter(2)) and result = 2` feels more QL'y to me.

Reply:
Agreed.  Updated.